test/framework/src/main/java/org/elasticsearch/cli/CommandTestCase.java

```diff
@@ -53,4 +53,15 @@ public String execute(String... args) throws Exception {
         command.mainWithoutErrorHandling(args, terminal);
         return terminal.getOutput();
     }
+
+    /**
+     * Runs the specified command with the given args.
+     * <p>
+     * Output can be found in {@link #terminal}.
+     */
+    public String execute(Command command, String... args) throws Exception {
```

Comment:
Why not just call mainWithoutErrorHandling directly in the one case where we need this? Making that method public seems better than overloading this method for one test case.

Reply:
I don't have a strong preference and don't mind following your suggestion but could you explain your reasoning/thinking regarding: 

> Making that method public seems better than overloading this method for one test case.

for future reference?

Reply:
Sorry I misunderstood where this method existed. It looks like we don't actually need a `command` member of the test case class. I would remove that, and make the other execute method call this method with `newCommand()`. Then I'm ok with having another method that takes command.